Data/models/CRUDProtocols.swift

```diff
@@ -56,7 +56,20 @@ public extension Deletable where Self: NSManagedObject {
                 }
             } else {
                 let deleteRequest = NSBatchDeleteRequest(fetchRequest: request)
```

Comment:
Alternatively, to make it simpler we could just use regular context deletion and iterating over collection of objects.
At the moment there are no places in code where we store a lot of objects(except for browsing history) so batch deletion doesn't bring that much of speed improvement.